src/components/tree/tree.tsx

```diff
@@ -183,6 +183,10 @@ export const Tree = memo(
       [itemElements]
     )
 
+    const handleOnFocus = useCallback((event: React.FocusEvent<HTMLDivElement>) => {
+      setFocusedElement(event.target as HTMLElement)
```

Comment:
Good catch to find this bug!

It would be somewhat better to use React's `.currentTarget` property so we can avoid type casting.

And we also need to make sure the `onFocus` property which can potentially be passed to `Tree` is called.

```suggestion
      setFocusedElement(event.currentTarget)
      
      // Call the element's `focus` handler
      onFocus?.(event)
```

Reply:
Hmm replacing with currentTarget will point me towards the TreeItem and not the Tree component which will then break the behavior again :/
That being said, when adding the rest I noticed that I don't need to cast 👍